source/opt/log.h

```diff
@@ -39,6 +39,21 @@
 #define SPIRV_ASSERT(consumer, ...)
 #endif
 
+// Logs an error message to consumer. Accepts the following formats:
```

Comment:
This is a lot of machinery to just delegate to spvtools::Log or spvtools::Logf but with one parameter (the message level substituted).  Why not have spvtools::LogError and spvtools::LogErrorF that just do that?  A using declaration to avoid typing the namespace would keep it short.